mayastor/src/bdev/nexus/nexus_io.rs

```diff
@@ -333,18 +332,16 @@ impl Bio {
                         child,
                     );
 
-                    let uri = child.name.clone();
                     nexus.pause().await.unwrap();
                     nexus.reconfigure(DrEvent::ChildFault).await;
-                    //nexus.remove_child(&uri).await.unwrap();
 
                     // Note, an error can occur here if a separate task,
                     // e.g. grpc request is also deleting the child,
                     // in which case the bdev may no longer exist at
                     // this point. To be addressed by CAS-632 to
                     // improve synchronization.
-                    if let Err(err) = bdev_destroy(&uri).await {
-                        error!("{} destroying bdev {}", err, uri)
+                    if let Err(err) = child.destroy().await {
```

Comment:
are there any other places where `child.set_state(ChildState::Destroying);` and `bdev_destroy` was used that could be replaced with this simple call?

Reply:
There are 2 calls to `bdev_destroy()` in `Nexus::add_child_only()`, the second of which could be replaced as it's after the `NexusChild` has been created but it's not clear how to reproduce that in a test.